### PR TITLE
[#77] Release 0.2.0

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Add sensible notes for the reviewer
1. [x] PR title is short, expressive and meaningful
1. [x] Relevant issues are linked
1. [x] Every source code file has a copyright header with `SPDX-License-Identifier: Apache-2.0 OR MIT`
1. [x] Branch follows the naming format (`iox2-123-introduce-posix-ipc-example`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`[#123] Add posix ipc example`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Assign PR to reviewer
1. [x] All checks have passed (except `task-list-completed`)

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/doc/release-notes/iceoryx2-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Unit tests have been written for new behavior
- [x] Public API is documented
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

Use either 'Closes #123' or 'Relates to #123' to reference the corresponding issue.

Relates to #77

### CODE REVIEW

File: doc/release-notes/iceoryx2-v0.2.0.md
Comment (elBoberido) [inline doc/release-notes/iceoryx2-v0.2.0.md:1]: What's your intention with `iceoryx2-unreleased.md`? Keep adding new stuff or starting from scratch?

**Review by elBoberido — DISMISSED** [review]

**Comment by elBoberido** [comment]
Oh, it seems cargo audit expects a released iceoryx2 v0.2

File: doc/release-notes/iceoryx2-v0.2.0.md
Comment (elfenpiff) [reply doc/release-notes/iceoryx2-v0.2.0.md:1]: Starting from scratch. I will reset it to the default empty values.

**Comment by elfenpiff** [comment]
@elBoberido mmhh it seems that we first have to release to crates.io and then merge the pull request. So I would split a release always in two PRs one for the docs and one PR that just adjust the version number. And this PR is already released to crates.io before it is merged.

**Comment by elBoberido** [comment]
> @elBoberido mmhh it seems that we first have to release to crates.io and then merge the pull request. So I would split a release always in two PRs one for the docs and one PR that just adjust the version number. And this PR is already released to crates.io before it is merged.

This doesn't sound right to me. I've read somewhere that tags should always be set on master and the the release should happen from the tag. Maybe we use cargo audit the wrong way.

**Comment by elBoberido** [comment]
@elfenpiff I think I know the solution but I'm not quite sure how fast I could implement it. It should be sufficient to run cargo audit only after the merge and for the PR have a check to ensure that the last cargo audit check passed. It's similar on how I ensured that a failure of the nightly build in iceoryx1 is noticed on new PRs.

**Comment by elfenpiff** [comment]
@elBoberido I think nearly all CI targets are affected by the problem, see the coverage and windows targets:
* https://github.com/eclipse-iceoryx/iceoryx2/actions/runs/7455143385/job/20283757389
* https://github.com/eclipse-iceoryx/iceoryx2/actions/runs/7455157283/job/20283785129

I would propose that we perform this release in the "uncleaner" way.
1. Release on crates.io
2. Merge `Cargo.toml` version change
3. Create Tag

And then we create a follow-up issue where we take a look at our CI, adjust the jobs so that we can release like:
1. Merge `Cargo.toml` version change
2. Create Tag
3. Release on crates.io

**Pull request merged**